spec/unit/facter/puppet_settings.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+# not sure how best to predict this? the following won't work on windows
+hostname = `hostname -f`.strip
+settings = {
+  "main" => {
+    "hostcert" => "/dev/null/ssl/certs/#{hostname}.pem",
```

Comment:
Technically Puppet uses the `certname` setting and not hostname. That just defaults to the hostname.